src/utils/routeHelpers.ts

```diff
@@ -161,10 +218,79 @@ const isAdmin = async (loggedUserID: number, neighborhoodID: number): Promise<bo
   return (neighborhood.admin_id === loggedUserID);
 };
 
+/**
+ * makes db query, fetches and returns associated fields to neighborhood
+ * adds admin, users and requests (to be implemented) fields and returns it
+ * throws error if neighborhood not present
+ * @param neighborhoodId
+ * @returns a promise resolved to neighborhood data with associated fields
+ */
+const generateNeighborhoodDataWithRelatedFields = async (neighborhoodId: number):
+Promise<NeighborhoodWithRelatedFields> => {
+  const dataWithRelatedFields: NeighborhoodWithRelatedFields | null = await prismaClient
+    .neighborhood.findFirst({
+      where: {
+        id: neighborhoodId,
+      },
+      include: {
+        admin: true,
+        users: true,
+        // requests: true, TODO: Radu please see why we cant include requests here
```

Comment:
I might be very wrong in this explanation since I do not have any docs that can back up my claims. 
By default Prisma generates types based on the default fields of the model (basically that means no relation fields such as admin, users and requests are included), however, for some reason, when you use `include` in a Prisma query, Prisma automatically generates a 'temporary' type just for that specific query and automatically assigns it to your const variable `dataWithRelatedFields`
Even though you created an interface in `types.ts` that basically matches the temporary type generated by Prisma, for some reason, Prisma does not want to recognize the fact that they are basically the same thing. So in order to make it work we have to play by the rules laid out by Prisma.
The solution is to replace your interface with this: 
```
const NeighborhoodWithRelatedFields = Prisma.validator<Prisma.NeighborhoodArgs>()({
  include: {
    users: true,
    requests: true,
  }
});

const NeighborhoodRelatedFields = Prisma.validator<Prisma.NeighborhoodArgs>()({
  select: {admin: true},
})

export type NeighborhoodWithRelatedFields = Prisma.NeighborhoodGetPayload<typeof NeighborhoodWithRelatedFields>;
```
https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety/operating-against-partial-structures-of-model-types#problem-using-variations-of-the-generated-model-type

I might be very wrong in my assumptions. This is what I think happens based on the error message provided. Let me know your opinion on this!

Reply:
They do work, I checked them before I wrote this comment. Do you want me to push them to the branch?

Reply:
I pushed the changes to the branch.